src/app/services/tokenService.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+angular
+  .module('jamjar')
+  .factory('TokenService', function($window) {
+    var service = this;
+
+    return {
+        setToken: function(token) {
+          $window.localStorage.token = token;
+        },
+
+        getToken: function() {
+          return $window.localStorage.token;
+        },
+
+        clearToken: function() {
+          delete $window.localStorage.token;
+        },
+
+        onUnauthorized: function() {
+          debugger
```

Comment:
This should probably route the user to the login page and wipe any local storage